specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,387 @@
+                }
+                else
+                {
+                    wil::com_ptr<ICoreWebView2StagingExecuteScriptException> exception;
+                    BOOL is_success;
+                    // The is_success will return false since an incorrect regular expression
+                    // was passed in.
+                    if (result->get_IsSuccess(&is_success) == S_OK && !is_success)
+                    {
+                        result->get_Exception(&exception);
+
+                        // Get the exception name, this could return the empty string,
+                        // such as `throw 1`.
+                        // In this case, we will get 'SyntaxError'.
+                        wil::unique_cotaskmem_string exceptionName;
+                        if (exception && exception->get_Name(&exceptionName) == S_OK)
+                        {
+                            MessageBox(
+                                nullptr, exceptionName.get(), L"ExecuteScript Result", MB_OK);
+                        }
+
+                        // Get the exception message, this could return the empty
+                        // string, such as `throw 1`.
+                        // In this case, we will get 'SyntaxError: Invalid regular expression:
+                        // missing /'.
+                        wil::unique_cotaskmem_string exceptionMessage;
+                        if (exception && exception->get_Message(&exceptionMessage) == S_OK)
+                        {
+                            MessageBox(
+                                nullptr, exceptionMessage.get(), L"ExecuteScript Result",
+                                MB_OK);
+                        }
+
+                        // Get the exception detail, it's a json struct data with all
+                        // exception infomation , we can parse it and get the detail
+                        // what we need.
+                        wil::unique_cotaskmem_string exceptionDetail;
+                        if (exception &&
+                            exception->get_ExceptionAsJSON(&exceptionDetail) == S_OK)
+                        {
+                            MessageBox(
+                                nullptr, exceptionDetail.get(), L"ExecuteScript Result", MB_OK);
+                        }
+
+
+                        // Get the location of the exception, note that the coordinates
+                        // here are 0 as the starting position.
+                        // In this case, we will get the coordinate: `LineNumber:0, ColumnNumber:55`.
+                        uint32_t lineNumber = 0;
+                        uint32_t columnNumber = 0;
+                        if (exception && exception->get_LineNumber(&lineNumber) == S_OK &&
+                            exception->get_ColumnNumber(&columnNumber) == S_OK)
+                        {
+                            auto exception_location_info =
```

Comment:
In a real app it might be nice to get the different properties from the exception and combine them into one string and do one MessageBox to display the error.